cpp/src/decisiontree/batched-levelalgo/kernels.cuh

```diff
@@ -250,6 +250,12 @@ __global__ void nodeSplitKernel(IdxT max_depth, IdxT min_rows_per_node,
                                              total_nodes, (char*)smem);
 }
 
+/* Returns 'input' rounded up to a correctly-aligned pointer of type OutT */
+template <typename OutT, typename InT>
+__device__ OutT alignPointer(InT input) {
+  return reinterpret_cast<OutT>(raft::alignTo((size_t)input, sizeof(OutT)));
```

Comment:
nitpick: can you use `reinterpret_cast` here?

Reply:
one more thing I noticed is that you'll need to align wrt the data-type, but here `sizeof(OutT)` means we are aligning wrt to its pointer's type!

Reply:
Yeah, I was thinking that pointer was correct, so this was by design, but I think you're right that it should be the real type.

Reply:
Fixed